Sources/SKSwiftPMWorkspace/SwiftPMWorkspace.swift

```diff
@@ -293,6 +293,13 @@ extension SwiftPMWorkspace: SKCore.BuildSystem {
 
     return nil
   }
+
+  public func filesDidChange(_ events: [FileEvent]) {
+    if events.contains(where: { $0.type == .created }) {
+      // TODO: It should not be necessary to reload the entire package just to get build settings for one file.
+      try? self.reloadPackage()
```

Comment:
Instead of `try?` you could wrap in `orLog` so that the failure is logged.

Reply:
Will things be in an invalid state if we set self.packageGraph but https://github.com/apple/sourcekit-lsp/blob/8503a3fc957118d6bf4b7f8ca687408e05048ddf/Sources/SKSwiftPMWorkspace/SwiftPMWorkspace.swift#L170  fails? Might want to set the packageGraph after that, Ben might know more.

Reply:
> Instead of `try?` you could wrap in `orLog` so that the failure is logged.

Oh, nice, I didn’t know about `orLog`. Thanks.

> Will things be in an invalid state if we set self.packageGraph but
> 
> https://github.com/apple/sourcekit-lsp/blob/8503a3fc957118d6bf4b7f8ca687408e05048ddf/Sources/SKSwiftPMWorkspace/SwiftPMWorkspace.swift#L170
> 
> fails? Might want to set the packageGraph after that, Ben might know more.

Good point. I think we should be safe if we create `BuildPlan` before setting `packageGraph`. I’ve flipped the two statement. What do you think?

Reply:
LGTM